topics/javascript/05-objects/06-practice/13-add-object-property/solution/addObjectProperty.js

```diff
@@ -1,7 +1,8 @@
-function addObjectProperty(obj1, key, obj2) {
-  /* eslint-disable-next-line no-param-reassign */
-  obj1[key] = obj2;
-  return obj2;
-}
+const addObjectProperty = (obj1, key, obj2) => {
+  const myObj2 = obj1;
```

Comment:
En este caso no hay necesidad de crear una nueva variable con una referencia al mismo objeto (cuando asignamos un objeto a una variable esto se hace por [referencia, no por valor](https://medium.com/laboratoria-how-to/por-valor-vs-por-referencia-en-javascript-de3daf53a8b9)), lo que significa que que no se _copia_ el objeto.

Por otro lado, el enunciado no hace referencia al valor de retorno, así que creo que esta solución se podría dejar así:

```js
const addObjectProperty = (obj1, key, obj2) => {
  obj1[key] = obj2;
};
```

O usando `Object.assign`:

```js
const addObjectProperty = (obj1, key, obj2) =>
  Object.assign(obj1, { [key]: obj2 });
```